src/vs/base/common/labels.ts

```diff
@@ -88,42 +88,82 @@ export function shorten(paths: string[]): string[] {
 	// for every path
 	let match = false;
 	for (let pathIndex = 0; pathIndex < paths.length; pathIndex++) {
-		const path = paths[pathIndex];
+		let path = paths[pathIndex];
+
+		if (path === '') {
+			shortenedPaths[pathIndex] = '.';
+			continue;
+		}
+
+		if (!path) {
+			shortenedPaths[pathIndex] = path;
+			continue;
+		}
+
 		match = true;
 
+		let prefix = '';
+		// trim for now and concatenate network path (e.g. \\network) or root path (/etc) later
+		if (path.indexOf(`${nativeSep + nativeSep}`) === 0) {
+			prefix = path.substr(0, path.indexOf(`${nativeSep + nativeSep}`) + 2);
+			path = path.substr(path.indexOf(`${nativeSep + nativeSep}`) + 2);
+		} else if (path.indexOf(`${nativeSep}`) === 0) {
+			prefix = path.substr(0, path.indexOf(`${nativeSep}`) + 1);
```

Comment:
Makes it unnecessarily hard to read to put a variable around ticks, just use `nativeSep` here?